src/core/source_buffers/queued_source_buffer.ts

```diff
@@ -269,153 +314,187 @@ export default class QueuedSourceBuffer<T> {
    * @private
    */
   public abort() {
-    this._buffer.abort();
+    this._sourceBuffer.abort();
   }
 
   /**
    * Callback used for the 'update' event, as a segment has been added/removed.
-   *
-   * Emit and complete the corresponding subject to inform the action caller
-   * of completion.
-   *
    * @private
    */
   private _onUpdate() : void {
-    if (this._flushing) {
-      this._flushing.next(undefined);
-
-      // security against side-effects from the previous `next` instruction
-      if (this._flushing) {
-        this._flushing.complete();
-        this._flushing = null;
-      }
+    if (!this.isLocked()) {
+      this._flush();
     }
   }
 
   /**
    * Callback used for the 'error' event from the SourceBuffer.
-   *
-   * Emit the error through the corresponding subject to inform the action
-   * caller.
-   *
    * @private
    * @param {Event} error
    */
   private _onError(error : Event) : void {
-    if (this._flushing) {
-      this._flushing.error(error);
-      this._flushing = null;
+    if (this._currentOrder != null) {
+      this._currentOrder.subject.error(error);
+    }
+  }
+
+  private _onUpdateEnd() : void {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 324. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=336&issues=AWbK5wRLSgN4uCclyMI9&open=AWbK5wRLSgN4uCclyMI9)

[ISSUE_KEY]: AWbK5wRLSgN4uCclyMI9